src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/mendrelation/PublicTransportMendRelationAction.java

```diff
@@ -1120,16 +1120,17 @@ boolean isRestricted(Way currentWay, Way previousWay, Node commonNode) {
             else if (rel.hasTag("type", "restriction") && rel.hasKey("restriction"))
                 return false;
             else {
-                boolean remove = true;
                 String routeValue = relation.get("route");
                 for (String s : restrictions) {
+                    if (s.length() <= 12)
+                        continue;
                     String sub = s.substring(12);
                     if (routeValue.equals(sub) && rel.hasTag("type", s))
```

Comment:
@michaelabon Sure, I was a bit in a rush earlier. This would be how it could be written:

```suggestion
                final String routeValue = Optional.ofNullable(relation.get("route")).map(it -> "restriction:" + it).orElse("");
                for (String s : restrictions) {
                    if (routeValue.equals(s) && rel.hasTag("type", s))
```

So previously the prefix `restriction:` was removed from `s` and then compared with `routeValue`.

But now we append the prefix `restriction:` to `routeValue` and then compare it with `s`.